iris-fitDataset/data.js

```diff
@@ -0,0 +1,128 @@
+  [5.8, 2.8, 5.1, 2.4, 2], [6.4, 3.2, 5.3, 2.3, 2], [6.5, 3.0, 5.5, 1.8, 2],
+  [7.7, 3.8, 6.7, 2.2, 2], [7.7, 2.6, 6.9, 2.3, 2], [6.0, 2.2, 5.0, 1.5, 2],
+  [6.9, 3.2, 5.7, 2.3, 2], [5.6, 2.8, 4.9, 2.0, 2], [7.7, 2.8, 6.7, 2.0, 2],
+  [6.3, 2.7, 4.9, 1.8, 2], [6.7, 3.3, 5.7, 2.1, 2], [7.2, 3.2, 6.0, 1.8, 2],
+  [6.2, 2.8, 4.8, 1.8, 2], [6.1, 3.0, 4.9, 1.8, 2], [6.4, 2.8, 5.6, 2.1, 2],
+  [7.2, 3.0, 5.8, 1.6, 2], [7.4, 2.8, 6.1, 1.9, 2], [7.9, 3.8, 6.4, 2.0, 2],
+  [6.4, 2.8, 5.6, 2.2, 2], [6.3, 2.8, 5.1, 1.5, 2], [6.1, 2.6, 5.6, 1.4, 2],
+  [7.7, 3.0, 6.1, 2.3, 2], [6.3, 3.4, 5.6, 2.4, 2], [6.4, 3.1, 5.5, 1.8, 2],
+  [6.0, 3.0, 4.8, 1.8, 2], [6.9, 3.1, 5.4, 2.1, 2], [6.7, 3.1, 5.6, 2.4, 2],
+  [6.9, 3.1, 5.1, 2.3, 2], [5.8, 2.7, 5.1, 1.9, 2], [6.8, 3.2, 5.9, 2.3, 2],
+  [6.7, 3.3, 5.7, 2.5, 2], [6.7, 3.0, 5.2, 2.3, 2], [6.3, 2.5, 5.0, 1.9, 2],
+  [6.5, 3.0, 5.2, 2.0, 2], [6.2, 3.4, 5.4, 2.3, 2], [5.9, 3.0, 5.1, 1.8, 2],
+];
+
+/**
+ * Converts an integer into its one-hot representation and returns
+ * the data as a JS Array.
+ */
+export function flatOneHot(idx) {
+  // TODO(bileschi): Remove 'Array.from' from here once tf.data supports typed
+  // arrays https://github.com/tensorflow/tfjs/issues/1041
+  // TODO(bileschi): Remove '.dataSync()' from here once tf.data supports
+  // datasets built from tensors.
+  // https://github.com/tensorflow/tfjs/issues/1046
+  return Array.from(tf.oneHot([idx], 3).dataSync());
+}
+
+/**
+ * Obtains Iris data, split into training and test sets and with the label
+ * converted into one-hot format.
+ *
+ * @param testSplit Fraction of the data at the end to split as test data: a
+ *   number between 0 and 1.
+ *
+ * @param returns A list of two datasets, [trainingData, testingData].
+ *   The datasets represent a shuffled partition of the raw IRIS data.
+ *   Elements of the yielded data will consist of [Features, Labels].
+ *   - Features as a rank-1 `Tensor` of length-4 of numbers.
+ *   - Labels as a rank-1 `Tensor` in one-hot format.
+ */
+export async function getIrisData(testSplit) {
+  // TODO(bileschi): Update shuffle etc. to use the tf.data API calls once
+  // it is possible to cache the results for performance and train-test split
+  // stability across epochs.
+
+  // Shuffle a copy of the raw data.
+  const shuffled = IRIS_RAW_DATA.slice();
+  tf.util.shuffle(shuffled);
+  // Split the data into training and testing portions.
+  const numTestExamples = Math.round(IRIS_RAW_DATA.length * testSplit);
+  const numTrainExamples = IRIS_RAW_DATA.length - numTestExamples;
+  const train = shuffled.slice(0, numTrainExamples);
+  const test = shuffled.slice(numTrainExamples);
+  // Split the data into into X & y and apply feature mapping transformations
+  const trainX = tf.data.array(train.map(r => r.slice(0, 4)));
+  const testX = tf.data.array(test.map(r => r.slice(0, 4)));
+  const trainY = tf.data.array(train.map(r => flatOneHot(r[4])));
```

Comment:
See my comment above. Might be worth make `flatOneHot` work in a batched manner for efficiency.

Reply:
Or perhaps even better: is it possible to use `map()`?

Reply:
See above.